scipy/interpolate/_bsplines.py

```diff
@@ -314,6 +322,14 @@ def __call__(self, x, nu=0, extrapolate=None):
         x = np.asarray(x)
         x_shape, x_ndim = x.shape, x.ndim
         x = np.ascontiguousarray(x.ravel(), dtype=np.float_)
+
+        # With periodic extrapolation we map x to the segment
+        # (self.t[k], self.t[-k]).
+        if extrapolate == 'periodic':
```

Comment:
Well yes, but these are points excluded and in your code you excluded whole segments: `t[k] .. t[k+1]` and similar at the other end.

Also why did you decide that the interval is open?